File: src/main/java/com/github/wgzhao/dbquery/dto/QueryResult.java
package com.github.wgzhao.dbquery.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
@AllArgsConstructor
public class QueryResult implements Serializable {
    @Serial
    private static final long serialVersionUID = 691617930352197222L;

    private final String apiVersion = "1.0";
    private int status;
    private boolean success;
    private String message;
    private int total;
    private Map<String, Object> data;

    public QueryResult() {
        this.status = 400;
        this.success = false;
        this.data = Map.of("result", new ArrayList<>());
    }

    public QueryResult(String message) {
        this.status = 400;
        this.success = false;
        this.message = message;
        this.data = Map.of("result", new ArrayList<>());
    }
}

File: src/main/kotlin/com/github/wgzhao/dbquery/dto/QueryResult.kt
package com.github.wgzhao.dbquery.dto

data class QueryResult (
    val code: Int = 0,
    var message: String? = null,
    var total: Int = 0,
    val data: List<Map<String, Any>>? = null
) {
    companion object {
        // 静态方法构造统一返回结果
        fun <T> success(data: List<Map<String, Any>>?): QueryResult {
            return QueryResult(0, "success", data?.size, data)
        }

        fun <T> error(code: Int, message: String?): ApiResponse<T?> {
            return ApiResponse<T?>(code, message, null)
        }
    }
}